pallets/handles/src/lib.rs

```diff
@@ -352,4 +271,132 @@ pub mod pallet {
+		/// # Returns
+		/// * `Vec<u16>` - The suffixes for the specified `base_handle`
+		pub fn get_next_suffixes(handle: Vec<u8>, count: u16) -> Vec<HandleSuffix> {
+			let mut suffixes: Vec<u16> = vec![];
+			let base_handle: Handle = handle.try_into().unwrap_or_default();
+			let base_handle_str = core::str::from_utf8(&base_handle).unwrap_or("");
+			let canonical_handle: Handle = Self::convert_to_canonical(base_handle_str);
+			let suffix_index =
+				Self::get_next_suffix_index_for_canonical_handle(canonical_handle.clone())
+					.unwrap_or_default();
+
+			let canonical_handle_str = core::str::from_utf8(&canonical_handle).unwrap_or("");
+			// Generate suffixes from the next available suffix index
+			let mut suffix_generator = SuffixGenerator::new(
+				T::HandleSuffixMin::get() as usize,
+				T::HandleSuffixMax::get() as usize,
+				&canonical_handle_str,
+			);
+			let sequence = suffix_generator.suffix_iter().enumerate();
+
+			for (i, suffix) in sequence {
+				if i >= suffix_index as usize && i < (suffix_index as usize + count as usize) {
+					suffixes.push(suffix as u16);
+				}
+				if suffixes.len() == count as usize {
+					break
+				}
+			}
+
+			suffixes
+		}
+
+		/// Get the next index into the shuffled suffix sequence for the specified canonical base handle
+		///
+		/// # Errors
+		/// * [`Error::SuffixesExhausted`]
+		///
+		pub fn get_next_suffix_index_for_canonical_handle(
+			canonical_handle: Handle,
+		) -> Result<SequenceIndex, DispatchError> {
+			let next;
+			match Self::get_current_suffix_index_for_canonical_handle(canonical_handle) {
+				None => {
+					next = 0;
+				},
+				Some(current) => {
+					next = current.checked_add(1).ok_or(Error::<T>::SuffixesExhausted)?;
+				},
+			}
+
+			Ok(next)
+		}
+
+		/// Verify the `signature` was signed by `signer` on `payload` by a wallet
+		/// Note the `wrap_binary_data` follows the Polkadot wallet pattern of wrapping with `<Byte>` tags.
+		///
+		/// # Errors
+		/// * [`Error::InvalidSignature`]
+		///
+		pub fn verify_signed_payload(
+			signature: &MultiSignature,
+			signer: &T::AccountId,
+			payload: Vec<u8>,
+		) -> DispatchResult {
+			let key = T::ConvertIntoAccountId32::convert((*signer).clone());
+			let wrapped_payload = wrap_binary_data(payload);
+
+			ensure!(signature.verify(&wrapped_payload[..], &key), Error::<T>::InvalidSignature);
+
+			Ok(())
+		}
+
+		fn convert_to_canonical(base_handle_str: &str) -> Handle {
+			let canonical_converter = CanonicalConverter::new();
+			let canonical_handle_str = canonical_converter.to_canonical(&base_handle_str);
+			let no_confusables_str = canonical_converter.remove_confusables(&canonical_handle_str);
+			let no_diacriticals_str = canonical_converter.strip_diacriticals(&no_confusables_str);
+			let canonical_handle_vec = no_diacriticals_str.as_bytes().to_vec();
+
+			let canonical_handle: Handle = canonical_handle_vec.try_into().unwrap();
+			canonical_handle
+		}
```

Comment:
Moved all canonical stuff from extrinsic and into this function as it is used in RPCs